components/automate-deployment/pkg/server/server.go

```diff
@@ -1761,6 +1761,45 @@ func (s *server) ManifestVersion(ctx context.Context, d *api.ManifestVersionRequ
 	}, nil
 }
 
+// SetManifest sets the current package manifest
+func (s *server) SetManifest(ctx context.Context, req *api.SetManifestRequest) (*api.SetManifestResponse, error) {
+	if !s.HasConfiguredDeployment() {
+		return nil, ErrorNotConfigured
+	}
+
+	var (
+		err             error
+		manifestClient  manifest.ReleaseManifestProvider
+		desiredManifest *manifest.A2
+		jsonManifest    []byte
+	)
+
+	jsonManifest = req.GetManifest().GetJson()
+	if jsonManifest == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse the manifest json")
+	}
+	manifestClient = client.NewInMemoryClient(jsonManifest)
+	desiredManifest, err = manifestClient.GetCurrentManifest(ctx, "")
+
+	if err == manifest.ErrCannotParse {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse the manifest")
+	} else if err != nil {
+		return nil, err
+	}
+
+	if err = s.acquireLock(ctx); err != nil {
```

Comment:
I put a comment on acquireLock that is worth a read to see if you really want to use it.

Reply:
I think it's fine, it's same thing as Lock(), it'll just check my context first.